dask/array/utils.py

```diff
@@ -71,12 +75,50 @@ def meta_from_array(x, ndim=None, dtype=None):
     except Exception:
         meta = np.empty((0,) * ndim, dtype=dtype or x.dtype)
 
+    if np.isscalar(meta):
+        meta = np.array(meta)
+
     if dtype and meta.dtype != dtype:
         meta = meta.astype(dtype)
 
     return meta
 
 
+def compute_meta(func, _dtype, *args, **kwargs):
+    args_meta = list(map(meta_from_array, args))
+    kwargs_meta = {k: meta_from_array(v) if is_arraylike(v) else v for k, v in kwargs.items()}
+
+    # todo: look for alternative to this, causes issues when using map_blocks()
+    # with np.vectorize, such as dask.array.routines._isnonzero_vec().
+    if isinstance(func, np.vectorize):
+        meta = func(*args_meta)
+        if _dtype and getattr(meta, 'dtype', None) != _dtype:
```

Comment:
This line and the two that follow look redundant, there's an exact copy of that starting on line 112 just below.

Reply:
Nice find.  Removed.